### ISSUE TITLE
DllNotFoundException

### ISSUE THREAD

**Issue by vaspotapov** [gilzoide/unity-texture-apply-async#1]
Hi, would like to test your approach, but I get this errors

```
DllNotFoundException: texture_async_apply assembly:<unknown assembly> type:<unknown type> member:(null)
Gilzoide.TextureApplyAsync.Internal.NativeBridge..cctor () (at Assets/Runtime/Internal/NativeBridge.cs:16)
Rethrow as TypeInitializationException: The type initializer for 'Gilzoide.TextureApplyAsync.Internal.NativeBridge' threw an exception.
Gilzoide.TextureApplyAsync.TextureApplyAsyncHandle..ctor (UnityEngine.Texture2D texture) (at Assets/Runtime/TextureApplyAsyncHandle.cs:26)
Test.Start () (at Assets/Test.cs:15)
```

how to resolve this?

**Comment by gilzoide** [comment]
Hey @vaspotapov! I should work as is on the supported platforms (Windows x86/x86_64, Linux x86_64, macOS x86/arm64, Android arm32/arm64/x86/x86_64, iOS, tvOS, visionOS). What platform are you trying to use this from?

Now, specifically if you are using macOS, `libtexture_async_apply.dylib` is not codesigned, which means any macOS systems, apart from the one I made the build with, don't trust this executable file and you have to either:
1. Codesign the dylib yourself. This way not only you, but other people can use the dylib without the system blocking its execution. (I'm not sure if you need to build it to be able to codesign it, but well, I'm no expert in macOS stuff)
2. Rebuild the dylib yourself without codesigning. You will be able to execute the dylib, since you built it in your machine, but other people running built apps will have the same problem, though. To build it, make sure you have XCode command line tools installed and run `make macos-universal` from the Plugins directory.
3. You can tell the system that you trust this .dylib file by manually opening it from Finder with a right click on it and choosing the "Open" context menu. The system will give you a big warning and there you can choose like "I know what I'm doing, open anyway".

**Comment by AndreiUsoltsev** [comment]
The same problem. Windows platform, Mono build backend, .Net Framework, Unity 2022.3

**Comment by gilzoide** [comment]
> Windows platform

Are you by any chance using Windows on an ARM64 machine? It's expected to not work on ARM64 because there are no native builds for this architecture yet.

Also, do you folks have this problem on the Unity editor, on builds or both?

**Comment by AndreiUsoltsev** [comment]
> > Windows platform
> 
> Are you by any chance using Windows on an ARM64 machine? It's expected to not work on ARM64 because there are no native builds for this architecture yet.
> 
> Also, do you folks have this problem on the Unity editor, on builds or both?

ARM64 is not used. The problem is reproduced both in the editor and in the build.

Logs from build:
`
DllNotFoundException: Unable to load DLL 'texture_async_apply'. Tried the load the following dynamic libraries: Unable to load dynamic library 'texture_async_apply' because of 'Failed to open the requested dynamic library (0x06000000) - Не найден указанный модуль.  (WinError:0000007e)
  at Gilzoide.TextureApplyAsync.Internal.NativeBridge..cctor () [0x00000] in <00000000000000000000000000000000>:0 
  at Gilzoide.TextureApplyAsync.TextureApplyAsyncHandle..ctor (UnityEngine.Texture2D texture) [0x00000] in <00000000000000000000000000000000>:0 `

**Comment by gilzoide** [comment]
Sorry about that, the Windows DLLs were being build with the wrong link flags when using MinGW.
It should work now, I released version 1.0.0-preview3 with this fix.
If you still have problems, feel free to reopen this issue.

Cheers \o/
